config/vufind/config.ini

```diff
@@ -1763,10 +1763,27 @@ hide_holdings[] = "World Wide Web"
 ;
 ; Available options:
 ;    Channels - Display links to channels of content related to record
+;    Bookplate - Display a bookplate or something similar
 ;    Similar - Similarity based on Solr lookup
 ;    WorldCatSimilar - Similarity based on WorldCat lookup
 related[] = "Similar"
 
+; URLs for bookplates if enabled in the related config above. %s represents an
+; image name or donor code that can be used to identify an image. This will
+; come from a Solr field (donor_code by default).
+; bookplate_full = https://your-institution.edu/bookplates/%s-full.jpg
+; bookplate_thumb = https://your-institution.edu/bookplates/%s-thumb.jpg
```

Comment:
Would it be more flexible to offer a set of predefined tokens, rather than a simple `%s` that always expects a specific value? For example, what if the user actually has full URLs for thumbnails and full-size images in their index? They can't express that here by just setting both bookplate_full and bookplate_thumb to %s. If we instead offered a few predefined tokens like %%title%% and %%img_name%% then there would be greater flexibility. I'm not proposing that we should make this super-complicated right now; we could start by just offering the simplest, bare minimum options... but setting up such a mechanism might make this a bit more extensible and flexible for the future.

Reply:
Fantastic suggestion! I did this in a _very_ simple way. I didn't really have to predefine tokens so much, rather, I divorced the thumbnail logic (previously image name) from the full image logic. This allows them to be _different_. I changed variable names and added a corresponding method to do this. I also setup the infrastructure to use "tokens" going forward. Now we have `%%img%%` and `%%thumb%%` both of which can be part of an image name (e.g. `donor_code` like we use) or the entire URL. If somebody wanted to do like we do, they could have a configuration similar to this:

```
bookplate_full = https://your-institution.edu/bookplates/%%img%%-full.jpg
```

But if someone indexes the full URL, they would simply have a configuration like this:

```
bookplate_full = %%img%%
```